src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -131,6 +131,13 @@ class CodeBlock private constructor(
     }
   }
 
+  /**
+   * Returns a copy of the code block with selected format parts replaced, similar to
+   * [java.lang.String.replaceAll].
+   */
+  internal fun replaceAll(oldValue: String, newValue: String) =
+      CodeBlock(formatParts.map { it.replace(oldValue, newValue) }, args)
```

Comment:
Once this completes, we could have a format part that is empty, or one that doesn’t match its arguments.

This is safe as an internal function but really dangerous in general. Want to add a warning to that effect?

Reply:
Nice catch, I'll definitely mention that in the comment.